techmatters/flex-plugins
# CHI-2614: Fix issue with the save and end button

## Description

- Bug fix: Error when clicking the `Save and end` button multiple times.

- Root cause: This bug can be reproduced only when trying to save online contact. This is because the `completeTask` `async` function takes more time to resolve while saving online contact. By the time `submitContactFormAsyncAction` gets resolved, the `loadingStatus` is updated to loaded and as such enables the "Save and End" button, which can now be clicked, thus causing the error.

- Fix: To fix this, we need to always set the `loadingStatus` to loading and keep the "Save and end" button in a loading state until `completeTask` gets resolved.

### Checklist

### Other Related Issues

None

### Verification steps

- Start a web chat
- Accept the task as a counsellor
- Fill in the necessary details and click on the `Save and end` button
- Verify that the `Save and end` button cannot be clicked a second time while contact is been saved.
